drivers/gpio/gpio_max14916.c

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (c) 2024 Analog Devices Inc.
+ * Copyright (c) 2024 Baylibre SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+
+#define LOG_LEVEL CONFIG_GPIO_LOG_LEVEL
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(gpio_max14916);
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#include "gpio_max14916.h"
+#include "gpio_max149x6.h"
+
+#define DT_DRV_COMPAT adi_max14916_gpio
+
+static int gpio_max14916_diag_chan_get(const struct device *dev);
+
+static int max14916_pars_spi_diag(const struct device *dev, uint8_t *rx_diag_buff, uint8_t rw)
+{
+	struct max14916_data *data = dev->data;
+	int ret = 0;
+
+	if (rx_diag_buff[0]) {
+		LOG_ERR("[DIAG] MAX14916 in SPI diag - error detected");
+
+		data->glob.interrupt.reg_bits.SHT_VDD_FLT = 0x1 & (rx_diag_buff[0] >> 5);
+		data->glob.interrupt.reg_bits.OW_ON_FLT = 0x1 & (rx_diag_buff[0] >> 4);
+		data->glob.interrupt.reg_bits.OW_OFF_FLT = 0x1 & (rx_diag_buff[0] >> 3);
+		data->glob.interrupt.reg_bits.CURR_LIM = 0x1 & (rx_diag_buff[0] >> 2);
+		data->glob.interrupt.reg_bits.OVER_LD_FLT = 0x1 & (rx_diag_buff[0] >> 1);
+
+		if (0x1 & (rx_diag_buff[0] >> 0))
+			LOG_ERR("[DIAG] MAX14916 in SPI diag - GLOBAL FAULT detected");
```

Comment:
Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

Reply:
Fixed